src/prefect/_internal/concurrency/services.py

```diff
@@ -91,6 +91,11 @@ def _stop(self, at_exit: bool = False):
             self._remove_instance()
             self._stopped = True
 
+            # Allow asyncio Task to be garbage-collected because it may
+            # contain references to all Prefect Task calls during a flow
+            # run.
+            del self._task
```

Comment:
I need to look at this again next week, not sure this is how we want to clean up the task.